src/main/kotlin/org/rust/lang/core/stubs/index/RsModulesIndex.kt

```diff
@@ -63,7 +64,11 @@ class RsModulesIndex : StringStubIndexExtension<RsModDeclItem>() {
         private fun key(mod: RsModDeclItem): String? {
             val pathAttribute = mod.pathAttribute
             return if (pathAttribute != null) {
-                FileUtil.getNameWithoutExtension(PathUtil.getFileName(pathAttribute))
+                val fileName = PathUtil.getFileName(pathAttribute)
+                if (fileName == RsConstants.MOD_RS_FILE)
+                    PathUtil.getParentPath(pathAttribute).split('/').last()
```

Comment:
I think `pathAttribute` can use `\` here, because it's a string from a source code, which in theory might be windows only. Let's use `getFileName` here as well? 

Or maybe convert the whole thing to `nio.Path`?

Reply:
I used `getFileName()` here.
I also adapted the `NameResolution.kt` to support Windows style paths (even on Linux systems).
Hope thats OK.